python/kvikio/nvcomp.py

```diff
@@ -79,147 +121,315 @@ def compress(self, data: cp.ndarray) -> cp.ndarray:
         # TODO: An option: check if incoming data size matches the size of the
         # last incoming data, and reuse temp and out buffer if so.
         data_size = data.size * data.itemsize
-        self.compress_temp_size = np.zeros((1,), dtype=np.int64)
-        self.compress_out_size = np.zeros((1,), dtype=np.int64)
-        self.compressor.configure(
-            data_size, self.compress_temp_size, self.compress_out_size
-        )
-        self.compress_temp_buffer = cp.zeros(self.compress_temp_size, dtype=np.uint8)
-        self.compress_out_buffer = cp.zeros(self.compress_out_size, dtype=np.uint8)
-        self.compressor.compress_async(
-            data,
-            data_size,
-            self.compress_temp_buffer,
-            self.compress_temp_size,
-            self.compress_out_buffer,
-            self.compress_out_size,
-            self.s.ptr,
+        self.config = self._manager.configure_compression(data_size)
+        self.compress_out_buffer = cp.zeros(
+            self.config["max_compressed_buffer_size"], dtype="uint8"
         )
-        return self.compress_out_buffer[: self.compress_out_size[0]]
+        size = self._manager.compress(data, self.compress_out_buffer)
```

Comment:
Would coerce objects going into to `Array`. Also would need to add `from kvikio._lib.arr import asarray`

```suggestion
        size = self._manager.compress(asarray(data), asarray(self.compress_out_buffer))
```